### ISSUE TITLE
[IOS] When the obscureText parameter of CupertinoTextField is set to true, the focus is requested, then the focus is lost, and the application crashes

### ISSUE THREAD

**Issue by changleibox** [flutter/flutter#77139]
## Steps to Reproduce

1. CupertinoTextField设置obscureText为true；
2. 点击输入框，请求焦点，键盘弹出，点击空白处失去焦点，键盘收起；
3. 程序崩溃；
4. 一次不行，就多重复以上步骤几次；

**Expected results:** 
程序正常运行

**Actual results:** 
程序崩溃

<details>
  <summary>Logs</summary>

```
[Bugly] Fatal signal(11) raised.
[  +23 ms] [Bugly]  Trapped fatal signal 'SIGSEGV(11)' 
           (
                0  ???                            0x0000000f2017378b 0x000000000000ffff + 0,
                1  Flutter                        0x00000001030a0963
_ZN7flutter19AccessibilityBridge15UpdateSemanticsENSt3__113unordered_mapIiNS_13SemanticsNodeENS1_4hashIiEENS1_8equal_toIiEENS1_9allocatorINS1_4pairIKiS3_EEEEEENS2_IiNS_25CustomAccessibilityActionES5_S7_NS8_INS9_ISA_SE_EEEEEE + 547,
                2  Flutter                        0x00000001030aac76
_ZN7flutter15PlatformViewIOS15UpdateSemanticsENSt3__113unordered_mapIiNS_13SemanticsNodeENS1_4hashIiEENS1_8equal_toIiEENS1_9allocatorINS1_4pairIKiS3_EEEEEENS2_IiNS_25CustomAccessibilityActionES5_S7_NS8_INS9_ISA_SE_EEEEEE + 80,
                3  Flutter                        0x0000000103443762
_ZNSt3__110__function6__funcIZN7flutter5Shell23OnEngineUpdateSemanticsENS_13unordered_mapIiNS2_13SemanticsNodeENS_4hashIiEENS_8equal_toIiEENS_9allocatorINS_4pairIKiS5_EEEEEENS4_IiNS2_25CustomAccessibilityActionES7_S9_NSA_INSB_ISC_SG_EEEEEEE4$_36NSA_ISK_EEFvvEEclEv + 108,
                4  F<…>
[+1005 ms] [Bugly] Forward signal(11) to the previous handler: 0x0
```

```
No issues found! (ran in 5.8s)
```

```
[✓] Flutter (Channel dev, 1.27.0-8.0.pre, on macOS 11.2.2 20D80 darwin-x64, locale zh-Hans-CN)
    • Flutter version 1.27.0-8.0.pre at /Users/changlei/Downloads/flutter
    • Framework revision b7d4806243 (12 days ago), 2021-02-19 09:22:45 -0800
    • Engine revision 6993cb229b
    • Dart version 2.13.0 (build 2.13.0-30.0.dev)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/changlei/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/changlei/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin version 203.7759

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (4 available)
    • Box (mobile)                                 • 22638e6260b59d12da776a809ee5fec9870aa041 • ios            • iOS 14.4
    • iPad Pro (11-inch) (2nd generation) (mobile) • 5957F313-57B2-4F38-BD84-C656437FC3E3     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)                              • macos                                    • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)                                 • chrome                                   • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
```

</details>

<details>
  <summary>Example</summary>

```dart
class __ContentState extends State<_Content> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('TextField设置obscureText=true，导致崩溃问题'),
      ),
      body: Container(
        padding: const EdgeInsets.all(10),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.end,
          children: <Widget>[
            const Expanded(
              child: CupertinoTextField(
                padding: EdgeInsets.symmetric(
                  vertical: 15,
                  horizontal: 15,
                ),
                obscureText: true,
              ),
            ),
            CupertinoButton(
              child: const Text('Set text'),
              onPressed: () {
                FocusManager.instance.primaryFocus?.unfocus();
              },
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

**Comment by TahaTesser** [comment]
Hi @changleibox 
Thanks for filing the issue, I can reproduce it on the `dev` and `master` channels.
Doesn't reproduce it on the `stable` and `beta`, 


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('TextField设置obscureText=true，导致崩溃问题'),
      ),
      body: Container(
        padding: const EdgeInsets.all(10),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.end,
          children: <Widget>[
            const Expanded(
              child: CupertinoTextField(
                padding: EdgeInsets.symmetric(
                  vertical: 15,
                  horizontal: 15,
                ),
                obscureText: true,
              ),
            ),
            CupertinoButton(
              child: const Text('Set text'),
              onPressed: () {
                FocusManager.instance.primaryFocus?.unfocus();
              },
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>


<details>
<summary>xcode logs</summary>

```bash
2021-03-03 18:38:35.609902+0530 Runner[35473:512851] Metal API Validation Enabled
2021-03-03 18:38:35.723976+0530 Runner[35473:512851] Warning: Unable to create restoration in progress marker file
2021-03-03 18:38:35.800441+0530 Runner[35473:513616] flutter: Observatory listening on http://127.0.0.1:61899/jIID3QSrgP4=/
2021-03-03 18:38:36.609002+0530 Runner[35473:513604] fopen failed for data file: errno = 2 (No such file or directory)
2021-03-03 18:38:36.609223+0530 Runner[35473:513604] Errors found! Invalidating cache...
2021-03-03 18:38:37.407559+0530 Runner[35473:513604] fopen failed for data file: errno = 2 (No such file or directory)
2021-03-03 18:38:37.407759+0530 Runner[35473:513604] Errors found! Invalidating cache...
(lldb) 
```

</details>



<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel dev, 1.27.0-8.0.pre, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 1.27.0-8.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision b7d4806243 (12 days ago), 2021-02-19 09:22:45 -0800
    • Engine revision 6993cb229b
    • Dart version 2.13.0 (build 2.13.0-30.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (4 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E            • ios            • iOS 14.4
    • iPhone 12 (mobile)   • 948EC106-78EE-4A69-8D31-83C00E6419DE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)      • macos                                • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)         • chrome                               • web-javascript • Google Chrome 88.0.4324.192

• No issues found!

```

```bash
[✓] Flutter (Channel master, 1.27.0-9.0.pre.176, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 1.27.0-9.0.pre.176 at /Users/tahatesser/Code/flutter_master
    • Framework revision dff42b8ee1 (4 hours ago), 2021-03-03 03:14:02 -0500
    • Engine revision cd0e54348b
    • Dart version 2.13.0 (build 2.13.0-93.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (4 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E            • ios            • iOS 14.4
    • iPhone 12 (mobile)   • 948EC106-78EE-4A69-8D31-83C00E6419DE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)      • macos                                • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)         • chrome                               • web-javascript • Google Chrome 88.0.4324.192

• No issues found!

```

</details>



Thank you

**Comment by LongCatIsLooong** [comment]
@chunhtai does this have to do with the recent semantics changes?

**Comment by chunhtai** [comment]
looking

**Comment by chunhtai** [comment]
confirmed this to be a regression caused by https://github.com/flutter/engine/commit/b3a18f5182827ca6db20f3e75bfe11b39368b4ec

**Comment by changleibox** [comment]
> Hi @changleibox
> Thanks for filing the issue, I can reproduce it on the `dev` and `master` channels.
> Doesn't reproduce it on the `stable` and `beta`
> 
> ### Steps to reproduce
> 1. Run the code sample below.
> 2. Focus on the `CupertinoTextField` and type any word "flutter`and then hit `Set text` button`
> 3. Observe the crash
> 
> Also crashes with `TextField`
> 
> code sample
> xcode logs
> flutter doctor -v
> Thank you

好的，谢谢你，期待早日修复这个bug。
